trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/CHandler.java

```diff
@@ -3598,14 +3597,14 @@ private List<Statement> handleLoopCondition(final ILocation loc, final IDispatch
 				mExprResultTransformer.transformSwitchRexIntToBool(condResult, loc, cond);
 		final List<Statement> result = new ArrayList<>(condTransformed.getStatements());
 		resultBuilder.addDeclarations(condTransformed.getDeclarations());
-		// Insert an if-statement: if (!cond) break;
+		// Insert an if-statement: if (cond) {} else break;

```

Comment:
Previosuly, we translated some loop `while (c) st` to `while (true) if (!c) break; st`. This led to inverted `ConditionAnnotation` and thus the validation of branching waypoints did not work as expected.
In the backtranslation`CACSL2BoogieBacktranslator` made already some attempt to invert the step infos again. However, this does not help us for the validation.

Reply:
However, I just checked the [backtranslation tests](https://jenkins.sopranium.de/job/Ultimate/job/Ultimate%20Nightly/job/PR-680/3/testReport/de.uni_freiburg.informatik.ultimate.regressiontest.backtranslation/CBacktranslationRegressionTestSuite/) and some of them fail now because of this change. We might get some more program states in the counterexamples (which can be actually helpful to extract assumptions 😉). Therefore we should check, if this deviation is acceptable (maybe our block encoding is now slightly "worse"?) and adapt the tests in that case.